### ISSUE TITLE
Runtime Fields: Support Enriching

### ISSUE THREAD

**Issue by nemhods** [elastic/elasticsearch#72899]
Something that's not possible right now with runtime fields, but I think would be very helpful:

The ability to use enrich policies with runtime fields. For example, I have an enrich policy that translates IPs to internal subnet names. It would be great to be able to apply this ad-hoc to any index that has internal IPs, using runtime fields.

This would allow a form of run-time joining of data using native Elasticsearch features, and would bring more feature parity with query languages like Splunks SPL or Microsofts Kusto.

Maybe more generically, the runtime fields feature could be extended to support applying any ingest processor at runtime, without having to reindex? If this was possible in a systematic way, runtime fields would immediately gain a lot of extra features.

**Comment by javanna** [comment]
We do have query time lookup/enrichment in the roadmap for runtime fields. We've been working on a couple of other tasks before we get to it but it's on our radar.

That said we don't have plans to allow users to apply ingest processors to runtime fields. With runtime fields, we have decided to use scripting as their definition, and we will continue down this path. While we have considered reusing ingest processors for this purpose, we have concluded that although there are similarities between the two there are also substantial differences which justify keeping ingest processors and runtime fields separate.
